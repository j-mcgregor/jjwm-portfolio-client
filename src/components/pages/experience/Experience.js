import React from 'react';
import PropTypes from 'prop-types';
import uuidv2 from 'uuid/v1';
import { ExperienceType } from '../../types/ExperienceType';
import ExpItem from './ExpItem';
import S from '../../styledComponents';

const Experience = ({ expList }) => {
  const list = expList && expList.map(e => <ExpItem {...e} key={uuidv2()} />);
  return (
    <S.Experience>
      <div className="listContainer">
        {list && list.length ? list : 'Please add an experience'}
      </div>
    </S.Experience>
  );
};

Experience.propTypes = {
  expList: PropTypes.arrayOf(ExperienceType)
};

Experience.defaultProps = {
  expList: []
};

export default Experience;
