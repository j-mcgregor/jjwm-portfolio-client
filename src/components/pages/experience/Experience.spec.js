import React from 'react';
import { shallow } from 'enzyme';
import Experience from './Experience';

describe('Experience', () => {
  it('should render a list if a list is present', () => {
    const props = {
      expList: [
        {
          company: 'Cardano Risk Management',
          companyUrl: 'https://www.cardano.co.uk/',
          city: 'London',
          role: 'Full-stack Javascript Developer',
          from: 'Jan 2019',
          to: 'Dec 2019',
          description: 'Lorem ipsum...'
        }
      ]
    };
    const wrapper = shallow(<Experience {...props} />);
    expect(wrapper.find('ExpItem')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render placeholder text if no list is present', () => {
    const props = {
      expList: []
    };
    const wrapper = shallow(<Experience {...props} />);
    expect(wrapper.find('ExpItem')).toHaveLength(0);
    expect(wrapper.find('.listContainer').text()).toBe(
      'Please add an experience'
    );
    // expect(wrapper).toMatchSnapshot();
  });
});
