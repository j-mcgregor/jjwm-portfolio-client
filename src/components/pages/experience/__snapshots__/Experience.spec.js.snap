// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience should render a list if a list is present 1`] = `ShallowWrapper {}`;
